### DISCUSSION TITLE
following turn based rpg tutorial, what does this mean and why am i getting this error

### DISCUSSION THREAD

**Discussion by qq735-personal** [HaxeFlixel/flixel#2619] [Q&A]
`Expected extends or implements or {diagnostics(4)`
![image](https://user-images.githubusercontent.com/49921877/183124740-83df16f8-e526-4d20-8f29-579516eff1c2.png)
my full code.
Im doing what the tutorial says (im on part 3) which is "For right now, we just want to get the Player class initialized, make sure that it works, and try adding it to our PlayState. It's not going to move or do anything yet, that will come in a minute. So save the changes to our Player class, and go back to the PlayState. We need to define our Player variable, so underneath the class line, add:
 var player:Player;"
Did that, error. help?

**Comment by filipp8** [comment]
you have to move your var declaration at line 6 inside the curly bracket at line 8
```
class PlayState extends FlxState
{
  var player:Payer;
  ...
}
```

**Reply by qq735-personal** [reply]
![image](https://user-images.githubusercontent.com/49921877/183127750-1da4d39b-880d-43a9-b483-b07ed335b013.png)
Now its just giving me more similar errors.

**Reply by qq735-personal** [reply]
ok, its just messing with me at this point. i didnt do a thing and the one of the errors fixed itself, and created another one.
![image](https://user-images.githubusercontent.com/49921877/183128643-0e80d339-dc41-4981-ad7d-94696f2e7946.png)

**Reply by Geokureli** [reply]
same problem as before but with the `create` function, should be 

 ```hx
 override public function create()
 {
    super.create()
    // code goes here
 }
 ```

{} brackets in general are used to group lines of code for a class, function, if, or for loop

> ok, its just messing with me at this point. i didn't do a thing and the one of the errors fixed itself

The errors don't usually go away as you change the code, you need to save and rebuild the project to see whether errors are fixed or if new ones were created

**Comment by filipp8** [comment]
you have to move the bracket at line 13 between line 9 and 10.
you are declaring a function in line 9 and then you have to open the bracket. What you find inside the brackets is the actual function behavior.
You should learn some general programming language syntax before...
